src/UI/toolbar/BatteryIndicator.qml

```diff
@@ -278,6 +322,117 @@ Item {
                         }
                     }
                 }
+                
+            }
+
+            SettingsGroupLayout {
+                heading:        qsTr("Battery State Display")
+                Layout.fillWidth: true
+                spacing: ScreenTools.defaultFontPixelWidth * 0.5
+                
+                RowLayout {
+                    //Layout.fillWidth: true
+                    
+                    QGCColoredImage {
+                        source: "/qmlimages/BatteryGreen.svg"
+                        height: ScreenTools.defaultFontPixelHeight * 5
+                        width: ScreenTools.defaultFontPixelWidth * 7
+                        fillMode: Image.PreserveAspectFit
+                        color: qgcPal.colorGreen
+                    }
+
+                    QGCLabel {
+                        Layout.preferredWidth: ScreenTools.defaultFontPixelWidth * 3.5 
+                        text: qsTr("100%")
+                        Layout.leftMargin: -13  
+                    }
+
+                    QGCColoredImage {
+                        source: "/qmlimages/BatteryYellowGreen.svg"
+                        height: ScreenTools.defaultFontPixelHeight * 5
+                        width: ScreenTools.defaultFontPixelWidth * 7
+                        fillMode: Image.PreserveAspectFit
+                        color: qgcPal.colorYellowGreen
+                    }
+
+                    FactTextField {
+                        id: threshold1Field
+                        fact: batterySettings.threshold1
+                        validator: IntValidator { bottom: 16 } // Value is at least 16
```

Comment:
This should be done by specifying min in the FactMetaData. Then you get automatic validation.

Reply:
I’ve implemented the validation fully on the C++ side in `BatteryIndicatorSettings.cc`. Now, in the QML, we simply use:

```qml
onEditingFinished: {
    batterySettings.setThreshold2(parseInt(text));
}
```